File: Models/Comment.cs
using VenatorWebApp.Models.Abstracts;
using VenatorWebApp.Models.Common;

namespace VenatorWebApp.Models
{
    public class Comment : Textual
    {
        //TODO: rework
        public TextualType ParentType { get; set; }
        //TODO: rework
        public int ParentId { get; set; }

        public override bool IsValid() => !string.IsNullOrEmpty(Text);
    }
}

File: ClientApp/src/app/model/comment.ts
import { TextualType } from "./enum/textual-type";
import { Publication } from "./publication";
import { Textual } from "./textual";

export class Comment extends Textual {
  parent!: Publication;
  parentType!: TextualType;
  likesCount!: number;
  dislikesCount!: number;
}
